### PR TITLE
Add code for flutter asgardeo integration app

### PR DESCRIPTION
**Pull request by Dinithi-Hasanika**
## Purpose
Sample flutter mobile application integrated with Asgardeo, which has the following features,

- User sign-up
- User sign-in
- Sign-out
- Secured External API calling
- View the user profile
- Edit the user profile

Git Issue:
https://github.com/wso2-enterprise/iam-engineering/issues/668

### CODE REVIEW

File: lib/configs/configs.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+
+const asgardeoLogo = "assets/images/asgardeo-logo-transparent.png";
+const flutterLogo = "assets/images/flutter5786.jpg";
+const repoURL = "https://github.com/Dinithi-Hasanika/Asgardeo-Flutter-APP";
+const gitHubLogo = "assets/images/GitHub-Mark.png";
+
+//styles
+const primaryColor = 0xffFF7300;
+const buttonRadius = 50.0;
```
Comment (Achintha444) [inline lib/configs/configs.dart]: Add an empty line. Check other files as well.

File: lib/configs/configs.dart [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const defaultPhotoURL = '<default-photo-url>';
+
+
+const asgardeoLogo = "assets/images/asgardeo-logo-transparent.png";
+const flutterLogo = "assets/images/flutter5786.jpg";
+const repoURL = "https://github.com/Dinithi-Hasanika/Asgardeo-Flutter-APP";
+const gitHubLogo = "assets/images/GitHub-Mark.png";
+
+//styles
+const primaryColor = 0xffFF7300;
```
Comment (Achintha444) [inline lib/configs/configs.dart]: Also we can create a theme to manage primary colours. Not mandatory.
https://docs.flutter.dev/cookbook/design/themes

File: lib/configs/configs.dart [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const clientId = '<client_id>';
```
Comment (Achintha444) [inline lib/configs/configs.dart]: Better to mention types of the variables. If it is a lot of work do it gradually. Not mandatory for this PR.

File: lib/configs/end_point_urls.dart [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import 'configs.dart';
+
+const discoveryUrl =
```
Comment (Achintha444) [inline lib/configs/end_point_urls.dart]: Keep a consistent way to manage constants. Either keep it like this, or use a class do not use both.

File: lib/pages/my_asgardeo_app.dart [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+          leadingWidth: 280,
+          leading:  Padding(
+            padding: const EdgeInsets.only(left: 116, right: 2,),
+            child: isUserLoggedIn ? Image.asset(
+              configs.asgardeoLogo,
+              scale:0.1,
+            ): const Text(''),
+          ),
+        ),
+        body: isUserLoggedIn
```
Comment (Achintha444) [inline lib/pages/my_asgardeo_app.dart:39]: Can we make this more readable. Like use switch ?

File: lib/utils/api_client.dart [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+import 'http_client.dart';
+import '../providers/page.dart';
+import '../providers/user.dart';
+import '../providers/user_session.dart';
+import '../utils/auth_client.dart';
+import '../utils/util.dart';
+
+final logger = Logger();
+
+class APIClient{
```
Comment (Achintha444) [inline lib/utils/api_client.dart]: Better to move the files related to API / https to another folder, for better management.

File: lib/configs/configs.dart (hunk 1 above)
Comment (Dinithi-Hasanika) [reply lib/configs/configs.dart]: added empty lines in all the files

File: lib/configs/configs.dart (hunk 2 above)
Comment (Dinithi-Hasanika) [reply lib/configs/configs.dart]: theme is included, primary color is able to change as per user's choice

File: lib/configs/configs.dart (hunk 3 above)
Comment (Dinithi-Hasanika) [reply lib/configs/configs.dart]: fixed

File: lib/configs/end_point_urls.dart (hunk 4 above)
Comment (Dinithi-Hasanika) [reply lib/configs/end_point_urls.dart]: fixed and used class

File: lib/pages/my_asgardeo_app.dart (hunk 5 above)
Comment (Dinithi-Hasanika) [reply lib/pages/my_asgardeo_app.dart:39]: switch is not working here to render widget hence conditional statement used

File: lib/utils/api_client.dart (hunk 6 above)
Comment (Dinithi-Hasanika) [reply lib/utils/api_client.dart]: fixed

File: lib/configs/configs.dart [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+
+const String asgardeoLogo = "assets/images/asgardeo-logo-transparent.png";
+const String flutterLogo = "assets/images/flutter5786.jpg";
+const String repoURL = "https://github.com/wso2/asgardeo-flutter-b2c-sample-app";
+const String gitHubLogo = "assets/images/GitHub-Mark.png";
+
+//styles
+const int primaryColor = 0xffFF7300;
+const double buttonRadius = 50.0;
```
Comment (brionmario) [inline lib/configs/configs.dart]:
```suggestion
const double buttonRadius = 50.0;

```

**Review by brionmario — DISMISSED** [review]

**Review by brionmario — APPROVED** [review]

**Review by dasuni-30 — APPROVED** [review]

**Pull request merged**